server/src/main/java/org/elasticsearch/cluster/coordination/StartJoinRequest.java

```diff
@@ -16,34 +16,37 @@
 
 /**
  * Represents the action of requesting a join vote (see {@link Join}) from a node.
- * The source node represents the node that is asking for join votes.
+ *
+ * A node running for election as the new master in a new term will send requests
```

Comment:
Suggest being explicit about which requests are being sent here:

```suggestion
 * A node running for election as the new master in a new term will broadcast a `{@link StartJoinRequest}`
```